compiler/lib/global_deadcode.mli

```diff
@@ -0,0 +1,21 @@
+(* Js_of_ocaml compiler
+ * http://www.ocsigen.org/js_of_ocaml/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, with linking exception;
+ * either version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ *)
+
+val add_sentinal : Code.program -> Code.program * Code.Var.t
```

Comment:
Do we really need `add_sentinal` to be exposed ? Can't it be done inside `f` directly ?

Reply:
It can't be done inside `f`, since `Deadcode.f` has to run after `Global_flow.f` runs, but the sentinal needs to be added before global flow runs (otherwise it messes up the index math when accessing variable arrays produced by global flow). This is the order that we have in `exact_calls` (which is a bit of a confusing place to run this, but it's where global flow is right now and unwinding these dependencies would require more refactoring):

```ocaml
let p, deadcode_sentinal = Global_deadcode.add_sentinal p in
let info = Global_flow.f ~fast p in
let p = Global_deadcode.f p deadcode_sentinal info in
p, info, Some deadcode_sentinal
```

Maybe this dependency could be documented better?

Reply:
Are you sure this is true ?

When I apply the following diff, the testsuite still run fine
```diff
diff --git a/compiler/lib/driver.ml b/compiler/lib/driver.ml
index 172bdcb324..6a481e3508 100644
--- a/compiler/lib/driver.ml
+++ b/compiler/lib/driver.ml
@@ -107,8 +107,8 @@ let exact_calls profile p =
     let p, info, deadcode_sentinal =
       if Config.Flag.globaldeadcode ()
       then
-        let p, deadcode_sentinal = Global_deadcode.add_sentinal p in
         let info = Global_flow.f ~fast p in
+        let p, deadcode_sentinal = Global_deadcode.add_sentinal p in
         let p = Global_deadcode.f p deadcode_sentinal info in
         p, info, deadcode_sentinal
       else
```

I don't expect your pass to access global_flow for the sentinal.

Or maybe one just need to create/allocate the sentinal before
```ocaml
        let deadcode_sentinal = Code.Var.fresh () in
        let info = Global_flow.f ~fast p in
        Global_deadcode.f p deadcode_sentinal info in
```

Reply:
No, you're right. I think this dependency existed in an earlier version due to a bug that has since been fixed. I changed this in https://github.com/ocsigen/js_of_ocaml/pull/1503/commits/07c37bdcb8dfa0229c30b42e5ca28aea47b7bee3.